synapse/app/homeserver.py

```diff
@@ -193,6 +193,7 @@ def _configure_named_resource(self, name, compress=False):
                     "/_matrix/client/versions": client_resource,
                     "/.well-known/matrix/client": WellKnownResource(self),
                     "/_synapse/admin": AdminRestResource(self),
+                    "/_synapse/client": client_resource,
```

Comment:
I know it's not part of the specced api, but I'd be tempted to put the new endpoint under `/_matrix/client`, for a couple of reasons:

 * we don't tell people to forward `/_synapse` through their reverse proxies. This is partly deliberate (to stop them exposing the admin api accidentally), but even if we did recommend it, people might forget it and not realise their mistake until password reset mysteriously 404s.
 * sticking a `client_resource` here means that you can do all the regular C/S API requests via `_synapse` - and once you can, people *will*, and then we and other HS implementers will have to support it forever.

Note that the existing `/submit_token` API is under `/_matrix`, for the same reasons.

Reply:
Good points. I seemed to remember that putting `/submit_token` under `/_matrix/client/unstable` was a mistake, but it might've been more the `unstable` bit that was regrettable, rather than `/_matrix/client`. Your other points spell out perfectly why `/_synapse` isn't the best idea though :slightly_smiling_face:

Reply:
I've changed it from `/_synapse/client/account/v1/password_reset/email/submit_token_confirm` to `/_matrix/client/unstable/password_reset/email/submit_token_confirm`.

Reply:
I'm unsure whether we still want to keep with `unstable`, however I don't see a better option here at the moment.

Reply:
Maybe even something like `/_matrix/client/synapse/...`.